YoCode/FrontEndCheck.cs

```diff
@@ -5,60 +5,98 @@
 using OpenQA.Selenium.Chrome;
 using OpenQA.Selenium.Support.UI;
 using System.IO;
+using Microsoft.Win32;
 
 namespace YoCode
 {
     internal class FrontEndCheck
     {
-        private readonly IWebDriver browser;
+        private static IWebDriver browser;
         private readonly string port;
+        private const string CHROME = "Google Chrome";
+        private const string FIREFOX = "Firefox";
+
+        public static bool running;
 
         public FrontEndCheck(string applicantsWebPort, string[] keyWords)
         {
+            if (string.IsNullOrEmpty(applicantsWebPort))
```

Comment:
I think this method needs to be refactored. There are too many branching statements, which makes it hard to follow.

A good place to start would be to extract the body of each of the branches to a named function (starting with the long if statements in the middle).

Maybe:
```
TryLaunchFirefox()
TryLaunchChrome()
```
I wonder if rather than if else branches, it should try to launch Firefox and if that fails, try to launch chrome?